content/docs/19-kubernetes-knowlege-hub/00-how-to/06-deploy-stateless-frontend-app.md

```diff
@@ -0,0 +1,208 @@
+
+```bash
+git clone https://github.com/Princesso/date-buddy.git
+```
+
+If you prefer to use a different stateless frontend app, you can do so. You may, however, get different results than in this tutorial. This tutorial only serves as a guide.
+
+## Create a Dockerfile on the App’s Root Directory.
+
+Before continuing this step, you must ensure that Docker runs on your machine. In the root directory of the app, create a file named **Dockerfile**.
+
+```bash
+touch Dockerfile
+```
+
+In a text editor, add the lines below to the Dockerfile.
+
+```bash
+FROM node:12
+
+WORKDIR /date-suggestions
+
+COPY package*.json ./
+
+RUN npm install
+
+COPY . .
+
+EXPOSE 3000
+
+CMD ["npm", "start"]
+```
+
+Also, create a **.dockerignore** file and add the following lines to it.
+
+```bash
+/node_modules
+/.pnp
+.pnp.js
+/coverage
+```
+
+## Build a Docker Image of the Application.
+
+This step packages the application into a portable image. To build the app’s image, run the Docker `build` command as shown:
+
+```bash
+docker build -t date-suggestions .
+```
+
+## Create a Kubernetes Deployment.
+
+Before continuing with this step, ensure that you have access to a Kubernetes cluster, as explained in the [prerequisites](https://www.notion.so/How-To-Deploy-A-Stateless-Frontend-App-with-Kubernetes-b885ae2307e94ef191a1b713fe29c81f). 
+
+In the application's root directory, create a Kubernetes Deployment file using the `kubectl` command below:
+
+```bash
+kubectl create deploy date-suggestions --image=date-suggestions --replicas=2 --port=3000 --dry-run=client -o yaml
+```
+
+The output of running the command above is a YAML representation of the deployment, likely similar to the lines below:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: date-suggestions
+spec:
+  selector:
+    matchLabels:
+      app: date-suggestions
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: date-suggestions
+    spec:
+      containers:
+        - name: date-suggestions
+          image: date-suggestions
+          ports:
+            - containerPort: 3000
+```
+
+ 
+modify the command to create the deployment, to add a command to pipe the output into a `deployment.yaml` file.
+
+```bash
+kubectl create deploy date-suggestions --image=date-suggestions --replicas=2 --port=3000 --dry-run=client -o yaml > deployment.yaml
+
+```
+
+Alternatively, you can use the `touch` command to create the `deployment.yaml` file, and then copy the YAML output from the command to create a deployment to it.
+
+```bash
+touch deployment.yaml
+```
+
+## Create a Kubernetes Service.
+
+Create and populate a Kubernetes Service file in the app's root directory. As the default, your application will only be accessible within the cluster. You'll need to create a Kubernetes service resource to expose it to the outside world. A service resource creates an abstraction over a set of pods that provides discovery and routing between them.
```

Comment:
```suggestion
Create and populate a Kubernetes Service file in the app's root directory. By default, your application will only be accessible within the cluster. You'll need to create a Kubernetes service resource to expose the application to resources outside the Kubernetes cluster. A service resource creates an abstraction over a set of pods that provides discovery and routing between them.
```
Outside world is a bit vague. Let's help the reader by being specific and stating outside of the cluster.